pyro_risks/models/score_v0.py

```diff
@@ -142,6 +143,36 @@ def add_lags(df, cols):
     return df
 
 
+def add_lags(df, cols):
+    """Add lags to dataframe df of the selected columns.
+
+    Lags added correspond to day -1, -3 and -7 and are added to each department separately.
+
+    Args:
+        df (pd.DataFrame): main dataframe with departement and day columns
+        cols (list of str): columns on which to add lags
+
+    Returns:
+        [type]: [description]
+    """
+    for var in cols:
+        for dep in df["departement"].unique():
+            tmp = df[df["departement"] == dep][["day", var]].set_index("day")
+            tmp1 = tmp.copy()
+            tmp1 = tmp1.join(
+                tmp.shift(periods=1, freq="D"), rsuffix="_lag1", how="left"
+            )
+            tmp1 = tmp1.join(
+                tmp.shift(periods=3, freq="D"), rsuffix="_lag3", how="left"
+            )
+            tmp1 = tmp1.join(
+                tmp.shift(periods=7, freq="D"), rsuffix="_lag7", how="left"
```

Comment:
maybe improve this with what @GHCamille suggested? 

> ts = pd.util.testing.makeTimeSeries(10000)
> ts.shift(5, 'D') # takes 0.5 sec
> ts.shift(5, np.timedelta64(1, 'D')) # takes 0.16 sec

Reply:
I trust the time computations, but maybe it's worth to be sure that this speeds up the thing ;-)

Reply:
These are the time stats with what is suggests here https://github.com/pandas-dev/pandas/issues/10475

<img width="1200" alt="image" src="https://user-images.githubusercontent.com/57073096/102329864-2b08d280-3f89-11eb-83a3-612201a87be1.png">

<img width="1139" alt="image" src="https://user-images.githubusercontent.com/57073096/102329908-3825c180-3f89-11eb-8bf8-cdec6f81d38a.png">

<img width="730" alt="image" src="https://user-images.githubusercontent.com/57073096/102329923-3b20b200-3f89-11eb-9483-a58d20b04cf5.png">

The thing that takes a lot of time is rather the way the `add_lags` function is constructed: the mutiple joins and the loop through all departements. This is something that is avoided with the function that I proposed, even it is probably not the bets we can do ...

Reply:
With the `add_lags_to_predict` function we can gain ~14 seconds:

<img width="1199" alt="image" src="https://user-images.githubusercontent.com/57073096/102330626-12e58300-3f8a-11eb-8c5d-d329046e2916.png">

<img width="735" alt="image" src="https://user-images.githubusercontent.com/57073096/102330629-1547dd00-3f8a-11eb-9842-fa39a88cca6b.png">

If you believe that it's not a real increase in performance, I'll close this PR, fine by me :)